### PR TITLE
Linter run on top of qualtiy-branch

### PR DESCRIPTION
**Pull request by CoMPaTech**
Linter had a lot of complaints, squashed most of them, leaving the below. Do not that this is on top of #24!

```
plugwise/message.py:14:47: E1101: Instance of 'PlugwiseMessage' has no 'args' member (no-member)
plugwise/message.py:15:14: E1101: Instance of 'PlugwiseMessage' has no 'ID' member (no-member)
plugwise/message.py:16:11: E1101: Instance of 'PlugwiseMessage' has no 'mac' member (no-member)
plugwise/message.py:17:19: E1101: Instance of 'PlugwiseMessage' has no 'mac' member (no-member)
plugwise/message.py:22:4: R0201: Method could be a function (no-self-use)
plugwise/node.py:80:4: R0201: Method could be a function (no-self-use)
plugwise/stick.py:1190:25: R1701: Consider merging these isinstance calls to isinstance(self.expected_responses[seq_id][1], (NodeInfoRequest, NodePingRequest)) (consider-merging-isinstance)
plugwise/stick.py:1245:23: R1701: Consider merging these isinstance calls to isinstance(self.expected_responses[seq_id][1], (NodeInfoRequest, NodePingRequest)) (consider-merging-isinstance)
plugwise/stick.py:1422:24: R1701: Consider merging these isinstance calls to isinstance(self._plugwise_nodes[mac], (PlugwiseCircle, PlugwiseCirclePlus, PlugwiseStealth)) (consider-merging-isinstance)
plugwise/stick.py:694:31: R1701: Consider merging these isinstance calls to isinstance(self.expected_responses[seq_id][1], (NodeAddRequest, StickInitRequest)) (consider-merging-isinstance)
plugwise/util.py:161:4: R0201: Method could be a function (no-self-use)
```

While leaving out all `C0121|W0231|C0115|R1702|W0703|C0116|W0707` checks - on a near future run we should also take a look at the (too) broad Exceptions and too-many-nested-blocks

### CODE REVIEW

**Comment by CoMPaTech** [comment]
#24 takes precedence and this one should be rebased after to more clearly depict the changes

Needs thorough testing!

**Comment by CoMPaTech** [comment]
Added `defusedxml over lxml` for Smiles

File: plugwise/connections/connection.py [hunk 1]
```diff
@@ -15,41 +11,46 @@
...
         self.stick = stick
         self.run_reader_thread = False
         self.run_writer_thread = False
         self._is_connected = False
         self._writer = None
 
+        self._reader_thread = None
+        self._write_queue = None
+        self._writer_thread = None
+
```
Comment (CoMPaTech) [inline plugwise/connections/connection.py:28]: Check if these inits are valid (having defaults is a linting must)

File: plugwise/connections/serial.py [hunk 2]
```diff
@@ -26,8 +20,10 @@ def __init__(self, port, stick=None):
         self._stopbits = STOPBITS
         self._parity = serial.PARITY_NONE
 
+        self._serial = None
+
```
Comment (CoMPaTech) [inline plugwise/connections/serial.py:24]: Check if these inits are valid (having defaults is a linting must)

File: plugwise/connections/socket.py [hunk 3]
```diff
@@ -27,8 +19,10 @@ def __init__(self, port, stick=None):
         self._socket_port = int(port_split[1])
         self._socket_address = (self._socket_host, self._socket_port)
 
+        self._socket = None
+
```
Comment (CoMPaTech) [inline plugwise/connections/socket.py:23]: Check if these inits are valid (having defaults is a linting must)

File: plugwise/node.py [hunk 4]
```diff
@@ -60,8 +55,8 @@ def __init__(self, mac, address, stick):
         self._firmware_version = None
         self._relay_state = False
         self._last_log_address = None
-        self._last_log_collected = False
-        self._last_info_message = None
+        self.last_log_collected = False
+        self.last_info_message = None
```
Comment (CoMPaTech) [inline plugwise/node.py:59]: These are used outside the class so they can't be private (hence `_` removed)

File: plugwise/node.py [hunk 5]
```diff
@@ -70,16 +65,16 @@ def get_node_type(self) -> str:
...
+
+            # Try again with reversed order
+            hw_model = HW_MODELS.get(
+                self._hardware_version[-2:]
+                + self._hardware_version[-4:-2]
+                + self._hardware_version[-6:-4],
+                None,
+            )
+            if hw_model:
+                return hw_model
```
Comment (CoMPaTech) [inline plugwise/node.py:77]: The infamous else after return

File: plugwise/nodes/circle_plus.py [hunk 6]
```diff
@@ -110,7 +105,7 @@ def _process_scan_response(self, message):
                                 CirclePlusScanRequest(self.mac, node_address)
                             )
                     break
-                elif node_address == 63:
+                if node_address == 63:
```
Comment (CoMPaTech) [inline plugwise/nodes/circle_plus.py:108]: if after break

File: plugwise/nodes/sed.py [hunk 7]
```diff
@@ -46,9 +39,11 @@ class NodeSED(PlugwiseNode):
     def __init__(self, mac, address, stick):
         super().__init__(mac, address, stick)
         self._SED_requests = {}
-        self._maintenance_interval = SED_MAINTENANCE_INTERVAL
+        self.maintenance_interval = SED_MAINTENANCE_INTERVAL
         self._new_maintenance_interval = None
 
+        self._wake_up_interval = None
+
```
Comment (CoMPaTech) [inline plugwise/nodes/sed.py:46]: Check if these inits are valid (having defaults is a linting must)

File: plugwise/smile.py [hunk 8]
```diff
@@ -7,7 +7,7 @@
 import aiohttp
 import async_timeout
 from dateutil.parser import parse
-from lxml import etree
+from defusedxml import ElementTree as etree
```
Comment (CoMPaTech) [inline plugwise/smile.py:10]: Experimental - though it seems to work fine (@CoMPaTech @bouwew)

File: plugwise/smile.py [hunk 9]
```diff
@@ -259,7 +259,7 @@ async def request(
         try:
             # Encode to ensure utf8 parsing
             xml = etree.XML(escape_illegal_xml_characters(result).encode())
-        except etree.XMLSyntaxError:
+        except etree.ParseError:
```
Comment (CoMPaTech) [inline plugwise/smile.py:262]: The only 'real' change between lxml and defusedxml (@CoMPaTech / @bouwew)

File: plugwise/smile.py [hunk 10]
```diff
@@ -794,10 +794,7 @@ def get_appliance_data(self, dev_id):
                     ):
                         continue
                     # The presence of either indicates a local active device, e.g. heat-pump or gas-fired heater
-                    if (
-                        measurement == "compressor_state"
-                        or measurement == "flame_state"
-                    ):
+                    if measurement in ["compressor_state", "flame_state"]:
```
Comment (CoMPaTech) [inline plugwise/smile.py:797]: Pylint complaint fixed

File: plugwise/stick.py [hunk 11]
```diff
@@ -139,6 +127,18 @@ def __init__(self, port, callback=None, print_progress=False):
...
+        self._nodes_discovered = None
+        self._receive_timeout_thread = None
+        self._run_watchdog = None
+        self._send_message_queue = None
+        self._send_message_thread = None
+        self._update_thread = None
+        self._watchdog_thread = None
+        self.init_callback = None
+        self.connection = None
+
```
Comment (CoMPaTech) [inline plugwise/stick.py:141]: Check if these inits are valid (having defaults is a linting must)

File: plugwise/stick.py [hunk 12]
```diff
@@ -295,7 +295,7 @@ def do_callback(self, callback_type, callback_arg=None):
     def _discover_after_scan(self):
         """ Helper to do callback for new node """
         node_discovered = None
-        for mac in self._nodes_not_discovered.keys():
+        for mac, _values in self._nodes_not_discovered:
```
Comment (CoMPaTech) [inline plugwise/stick.py:298]: Linter complaining unnecessary use of keys - added dummy value

File: plugwise/stick.py [hunk 13]
```diff
@@ -346,10 +346,8 @@ def discover_node(self, mac: str, callback=None, force_discover=False) -> bool:
                             callback,
                         )
                 return True
-            else:
-                return False
-        else:
             return False
+        return False
```
Comment (CoMPaTech) [inline plugwise/stick.py:350]: else after return

File: plugwise/stick.py [hunk 14]
```diff
@@ -465,10 +461,8 @@ def node_join(self, mac: str, callback=None) -> bool:
         if validate_mac(mac) is True:
             self.send(NodeAddRequest(bytes(mac, UTF8_DECODE), True), callback)
             return True
-        else:
-            _LOGGER.warning(
-                "Invalid mac '%s' address, unable to join node manually.", mac
-            )
+
+        _LOGGER.warning("Invalid mac '%s' address, unable to join node manually.", mac)
```
Comment (CoMPaTech) [inline plugwise/stick.py:465]: else after return

File: plugwise/stick.py [hunk 15]
```diff
@@ -479,10 +473,10 @@ def node_unjoin(self, mac: str, callback=None) -> bool:
...
             )
             return True
-        else:
-            _LOGGER.warning(
-                "Invalid mac '%s' address, unable to unjoin node manually.", mac
-            )
+
+        _LOGGER.warning(
+            "Invalid mac '%s' address, unable to unjoin node manually.", mac
+        )
```
Comment (CoMPaTech) [inline plugwise/stick.py:479]: else after return

File: plugwise/stick.py [hunk 16]
```diff
@@ -1534,7 +1528,7 @@ def _update_loop(self):  # noqa: C901
                         update_loop_checker += 1
 
         except Exception as e:
-            exc_type, exc_obj, exc_tb = sys.exc_info()
+            _exc_type, _exc_obj, exc_tb = sys.exc_info()
```
Comment (CoMPaTech) [inline plugwise/stick.py:1531]: dummy/privatize unused vars

File: tests/test_smile.py [hunk 17]
```diff
@@ -1297,7 +1295,7 @@ async def test_fail_legacy_system(self):
         """Test erroneous legacy stretch system."""
         self.smile_setup = "faulty_stretch"
         try:
-            server, smile, client = await self.connect_wrapper()
+            _server, _smile, _client = await self.connect_wrapper()
```
Comment (CoMPaTech) [inline tests/test_smile.py:1298]: privatized unused vars

**Comment by CoMPaTech** [comment]
@bouwew this one should be better readable now #24 is out of the way - some small remarks with our names on it, feel free to close them if in agreement (we should all be able to do that as owners).

@brefra I think most of this is for you to verify/validate. I've commented on most things I changed, so if they are ok just close/resolve the conversations (we should all be able to do that as owners).

In general:

1. the `self.whatever = None` in added in the respective `__init__`s. Per linting it's required all self's have a sane default initialized in init and these didnt.
2. I renamed some `_function_name` to `function_name` again as per lint, since `_` should only be available private in a class and not outside the class
3. Mostly removed `from plugwise.message import PlugwiseMessage` from all files since it wasn't used - also other CONSTS were removed when not used.

File: plugwise/smile.py (hunk 8 above)
Comment (bouwew) [reply plugwise/smile.py:10]: I can confirm.

File: plugwise/smile.py (hunk 9 above)
Comment (bouwew) [reply plugwise/smile.py:262]: Good find!

File: plugwise/smile.py (hunk 10 above)
Comment (bouwew) [reply plugwise/smile.py:797]: Nice improvement!

File: tests/test_smile.py (hunk 17 above)
Comment (bouwew) [reply tests/test_smile.py:1298]: OK

**Review by bouwew — COMMENTED** [review]
Please fix the `if something is True:` and `if something is False:` occurrences as well.

**Review by bouwew — APPROVED** [review]
OK!

File: plugwise/messages/responses.py [hunk 18]
```diff
@@ -64,7 +64,7 @@ def deserialize(self, response):
         response = self._parse_params(response)
         crc = response[:4]
```
Comment (CoMPaTech) [inline plugwise/messages/responses.py:65]: Than crc needs to go ... it's unused

**Review by brefra — APPROVED** [review]
Tested this branch and it works for me 👍

**Pull request merged**